### PR TITLE
Step by Step Instructions issue #1

### PR DESCRIPTION
**Pull request by nelsonic**
We have a fairly major gap in our collective knowledge with regards to Phoenix LiveView.
So this Pull Request is a step-by-step complete beginner's tutorial building a counter app in LiveView.
This is the simplest _possible_ example app with the smallest codebase. It's _perfect_ for learning all the concepts without overwhelming the reader with any complexity.

**`BEFORE`**: https://github.com/dwyl/phoenix-liveview-counter-tutorial/tree/20bea0b36f0a6a054b4398c5f69fe4a1583282c4
***`AFTER`***: https://github.com/dwyl/phoenix-liveview-counter-tutorial/tree/step-by-step-instructions-issue%231

### CODE REVIEW

**Comment by nelsonic** [comment]
phoenix-liveview-counter-single-windowl

**Comment by nelsonic** [comment]
phoenix-liveview-counter-two-windows-independent-count

**Comment by nelsonic** [comment]
Counter updates in all connected browsers:
phoenix-liveview-counter-four-windows

File: README.md [hunk 1]
```diff
@@ -1,23 +1,1377 @@
-# Phoenix LiveView Counter Tutorial [![Build Status](https://img.shields.io/travis/dwyl/phoenix-liveview-counter-tutorial/master.svg?style=flat-square)](https://travis-ci.org/dwyl/phoenix-liveview-counter-tutorial) [![contributions welcome](https://img.shields.io/badge/contributions-welcome-brightgreen.svg?style=flat-square)](https://github.com/dwyl/phoenix-liveview-counter-tutorial/issues) [![HitCount](http://hits.dwyl.io/dwyl/phoenix-liveview-counter-tutorial.svg)](http://hits.dwyl.io/dwyl/phoenix-liveview-counter-tutorial)
+<div align="center">
```
Comment (nelsonic) [inline README.md]: See: https://github.com/dwyl/phoenix-liveview-counter-tutorial/tree/step-by-step-instructions-issue%231 👀

**Comment by nelsonic** [comment]
@SimonLab this tutorial is _finished_! 🏁 
When you have time, please review the [`README.md`](https://github.com/dwyl/phoenix-liveview-counter-tutorial/tree/step-by-step-instructions-issue%231) 👀
It's quite long, but I wanted to be _comprehensive_ so _complete beginners_ could learn the fundamentals of LiveView as fast as possible.

Once the PR is merged we can submit it to https://elixirforum.com http://elixirstatus.com and HN. 👍

File: README.md [hunk 2]
```diff
@@ -1,23 +1,1377 @@
...
+
+> 💡 This is a **random** string generator
+that generates a **32 character string** of alphanumeric data, <br />
+so the result will be **different each time** you run the command.
+
+Copy the string into your computer's clipboard.
+
+
+Open your `config/config.exs` file
+and locate the line that begins with
```
Comment (SimonLab) [inline README.md]: I'm not sure the configuration `live_view` is added by default when the live_view dependency is installed.
In my case I had to add (not just replace) the value.

Comment (nelsonic) [reply README.md]: Ah, please ensure you're using Phoenix `1.4.15` 💭

File: README.md [hunk 3]
```diff
@@ -1,23 +1,1377 @@
...
+for the `signing_salt`
+to ensure it is kept secret.
+
+
+<br />
+
+### Step 4: Add `:fetch_live_flash` Plug to Browser Pipeline
+
+
+At the top of the file you should see the line
```
Comment (SimonLab) [inline README.md]: We need to specify which file as we are switching from the config.exs file to the router.ex file
```suggestion
At the top of the [router.ex](https://github.com/dwyl/phoenix-liveview-counter-tutorial/blob/3a5e5dd73c325ef609203e597cd0f80ac06198f8/lib/live_view_counter_web/router.ex#L2) file  you should see the line
```

Comment (nelsonic) [reply README.md]: Great point @SimonLab ✋

**Comment by nelsonic** [comment]
FYI: I'm aware that `phoenix_live_view 0.9.0` was released 2 days ago. 🎉 🙄 

When I attempt to update to use `v0.9.0` the App breaks!
image

I've spent a pomodoro trying to figure it out but not had any luck.
```sh
rm mix.lock
rm -rf _build
rm -rf deps
rm -rf assets/node_modules
mix deps.get
```
Output:
```
Dependency resolution completed:
New:
  cowboy 2.7.0
  cowlib 2.8.0
  file_system 0.2.8
  gettext 0.17.4
  jason 1.1.2
  mime 1.3.1
  phoenix 1.4.15
  phoenix_html 2.14.0
  phoenix_live_reload 1.2.1
  phoenix_live_view 0.9.0
  phoenix_pubsub 1.1.2
  plug 1.9.0
  plug_cowboy 2.1.2
  plug_crypto 1.1.2
  ranch 1.7.1
  telemetry 0.4.1
* Getting phoenix (Hex package)
* Getting phoenix_pubsub (Hex package)
* Getting phoenix_html (Hex package)
* Getting phoenix_live_reload (Hex package)
* Getting gettext (Hex package)
* Getting jason (Hex package)
* Getting plug_cowboy (Hex package)
* Getting phoenix_live_view (Hex package)
* Getting cowboy (Hex package)
* Getting plug (Hex package)
* Getting mime (Hex package)
* Getting plug_crypto (Hex package)
* Getting cowlib (Hex package)
* Getting ranch (Hex package)
* Getting file_system (Hex package)
* Getting telemetry (Hex package)
```
Install node deps:
```
npm install --prefix assets
```
Output: 
```
added 780 packages from 329 contributors and audited 8438 packages in 13.838s
```

```sh
mix phx.server
```

No luck! 😞 

I think we need just _ship_ the current PR 
and then work on the update to `0.9.0` _separately_: https://github.com/dwyl/phoenix-liveview-counter-tutorial/issues/5

File: README.md [hunk 4]
```diff
@@ -1,23 +1,1377 @@
...
+to communicate with the server,
+we need to open a socket.
+Open the
+[`lib/live_view_counter_web/endpoint.ex`](https://github.com/dwyl/phoenix-liveview-counter-tutorial/blob/3c7c448ead7c161167fb310638b44be80b20ea1e/lib/live_view_counter_web/endpoint.ex)
+file
+and add the following lines to it:
+
+```elixir
+socket "/live", Phoenix.LiveView.Socket,
+  websocket: [connect_info: [session: @session_options]]
```
Comment (SimonLab) [inline README.md]: Do we also need to indicate how to define the `@session_options`? It wasn't already defined for me (not sure if it is supposed to be added when the live view dependency is added)

File: README.md (hunk 2 above)
Comment (SimonLab) [reply README.md]: ok I'm using 1.4.11

File: README.md (hunk 4 above)
Comment (nelsonic) [reply README.md]: Oh, fair enough. When I created a new phoenix app (Step 1) and installed `phoenix_live_view` (Step 2) a lot of the setup was performed automagically ... I wonder if it's a Mac thing? 🤷‍♂ 

It appears we need to add a bit more from the official installation instructions:
https://hexdocs.pm/phoenix_live_view/installation.html

image

I can do this now.

File: README.md (hunk 2 above)
Comment (nelsonic) [reply README.md]: Either way I've added a section instructing people to add: 
```elixir
config :live_view_counter, LiveViewCounterWeb.Endpoint,
   live_view: [
     signing_salt: "iluKTpVJp8PgtRHYv1LSItNuQ1bLdR7c"
   ]
```
to their `config.exs` if they don't _already_ have a `live_view` config. 👍

File: README.md (hunk 4 above)
Comment (nelsonic) [reply README.md]: Section added: https://github.com/dwyl/phoenix-liveview-counter-tutorial/pull/4/commits/8b7fb02ea1587dd215ec59748a23b83a32a57336

Comment (SimonLab) [reply README.md]: I've updated my Phoenix to 1.4.15 and the `@session_options` is now defined when I installed the dependency

Comment (nelsonic) [reply README.md]: @SimonLab yeah, `1.4.15` appears to add some enhancements to post install. ✨ 
I think the aim of the Phoenix team is reduce the number of configuration steps to using LiveView.
With LiveView, Phoenix has the potential to _replace_ React (_or other front-end frameworks_) so they want/need to make the setup process as fast as possible. 💡

**Comment by nelsonic** [comment]
Thanks @SimonLab! 🥇

**Pull request merged**